apps/fortuna/src/keeper/process_event.rs

```diff
@@ -35,6 +38,12 @@ pub async fn process_event_with_backoff(
 
     metrics.requests.get_or_create(&account_label).inc();
     tracing::info!("Started processing event");
+    process_param.history.write().await.add(
+        (chain_state.id.clone(), event.sequence_number),
+        TimedJournalLog::with_current_time(JournalLog::Observed {
+            tx_hash: event.tx_hash,
+        }),
+    );
```

Comment:
maybe we should use a channel here to asynchronously send the events over to a separate thread that adds them to history. If we lock here, then someone could potentially down the keeper thread by spamming the `/v1/explorer` endpoint. If that happens, I'd rather drop events from the history than fail callbacks.

Reply:
one more thing to watch out for: this code will execute multiple times for every single sequence number, and in some cases the event information will differ. (specifically, if there's a chain reorg, then the same sequence number will represent two different events)